x-pack/plugins/cloud_security_posture/server/saved_objects/data_views.ts

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  ElasticsearchClient,
+  ISavedObjectsRepository,
+  SavedObject,
+  type KibanaRequest,
+  type Logger,
+} from '@kbn/core/server';
+import { DataViewAttributes } from '@kbn/data-views-plugin/common';
+import { SpacesServiceStart } from '@kbn/spaces-plugin/server';
+import { DataViewsServerPluginStart } from '@kbn/data-views-plugin/server';
+
+import {
+  CDR_MISCONFIGURATIONS_DATA_VIEW_ID_PREFIX,
+  CDR_MISCONFIGURATIONS_DATA_VIEW_NAME,
+  CDR_MISCONFIGURATIONS_INDEX_PATTERN,
+  CDR_VULNERABILITIES_DATA_VIEW_ID_PREFIX,
+  CDR_VULNERABILITIES_DATA_VIEW_NAME,
+  CDR_VULNERABILITIES_INDEX_PATTERN,
+} from '../../common/constants';
+
+const PREVIOUS_VERSION_INDEX_PATTERNS = {
+  [CDR_MISCONFIGURATIONS_DATA_VIEW_ID_PREFIX]: [],
+  [CDR_VULNERABILITIES_DATA_VIEW_ID_PREFIX]: [],
+};
+const CURRENT_VERSION_INDEX_PATTERNS: Record<string, string> = {
+  [CDR_MISCONFIGURATIONS_DATA_VIEW_ID_PREFIX]: CDR_MISCONFIGURATIONS_INDEX_PATTERN,
+  [CDR_VULNERABILITIES_DATA_VIEW_ID_PREFIX]: CDR_VULNERABILITIES_INDEX_PATTERN,
+};
+
+const DATA_VIEW_TIME_FIELD = '@timestamp';
+
+const getDataViewSafe = async (
+  soClient: ISavedObjectsRepository,
+  currentSpaceId: string,
+  currentSpaceDataViewId: string,
+  indexPattern: string,
+  logger: Logger
+): Promise<SavedObject<DataViewAttributes> | undefined> => {
+  try {
+    const dataView = await soClient.get<DataViewAttributes>(
+      'index-pattern',
+      currentSpaceDataViewId,
+      {
+        namespace: currentSpaceId,
+      }
+    );
+
+    // Verify user didn't edit the index pattern
+    const currentIndexPatterns = dataView.attributes.title.split(',');
+    const requiredIndexPatterns = indexPattern.split(',');
```

Comment:
another case we need to cover is us updating parts of the index pattern, eg.  from `logs-*latest_misconfigurations-cdr*` to `logs-*latest_misconfigurations-cdr*` or smth. Maybe it is worth a separate ticket to consider handling our updates to indexes so that users don't end up with messy index patterns. Eg. by doing 3 way merge: prevRequiredPattern, currentUserDefinedPattern, nextRequiredPattern. Again, the cross cluster search support might make it more complicated, as we would like to update the index pattern part without touching the remote cluster part which can be defined by user